test/test_service_runner.py

```diff
@@ -35,16 +30,21 @@ def warnings():
         yield patched
 
 
-class Service(object):
-    name = "service"
+@pytest.fixture
```

Comment:
So that it can have the `tracker` in scope, which is a fixture.

I originally changed it because I thought the `reset_mock` was responsible for the dodgy test, and only later discovered the duplicate event dispatch.

The "tracker" mock is the pattern we've used elsewhere to track calls like this though, so I left it.